### DISCUSSION TITLE
Zero intensity precursors in the output

### DISCUSSION THREAD

**Discussion by Mengbo-Li** [vdemichev/DiaNN#342] [Q&A]
Hi Vadim, 

May I ask in what cases will there be a zero intensity precursor? In the main report, I see a few precursors with very small Q values but all their precursor-level quantities are zero. I am currently just removing those entries from my analysis, but it still interests me at which processing steps those zeros may arise. 

Cheers, 
Mengbo

**Comment by vdemichev** [comment]
Hi Mengbo,

DIA-NN selects which fragments to use to quantify precursors in cross-run manner. Sometimes, all fragments selected have 0 quantities. This means that 0 is the best estimate for the precursor quantity. If this is not compatible with the downstream workflow, please just keep considering these missing values, that's also fine.

Best,
Vadim

**Reply by Mengbo-Li** [reply]
Thanks Vadim for your helpful reply. So DIA-NN does make a distinction between missing values (which are not output in the main report) from zero intensities is that right? In the Nature methods paper, it says "All the precursor intensities corresponding to identifications with q-values above a given threshold are replaced with zeros and preliminary cross-run normalization based on the total signal (that is, the sum of the intensities of all precursors) is performed. ", is this where those zero quantities originate from? Sorry that I am a bit confused as it might be my misunderstanding, but if the fragment has zero intensity, there should not be a peak in its chromatogram to match in the first place? I.e. such a fragment should just be absent from the run?
Thanks in advance, 
Mengbo

**Reply by vdemichev** [reply]
Hi Mengbo, no, the zeroes don't come from that procedure. 
Yes, not peak from fragment with zero intensity, but there are peaks from other fragments, which allow to identify the precursor. But since all 3 fragments used to quantify it are 0, the quantity will be 0.
